From 789d1ce8194a4e444ec6ee71b421d3bde0ed2231 Mon Sep 17 00:00:00 2001
From: "Bhakte, GurudattaX" <gurudattax.bhakte@intel.com>
Date: Wed, 27 May 2015 17:12:39 +0530
Subject: [PATCH] ASoC: Intel: rt5670: Turn Off the mic bias levels when
 headset is disconnected

Issue: GMINL-10304

On CHT-FFD-HR mic bias level was remaining On all the time after first headset
insertion, which causes clicks when inserting headset.
This patch turn Off the mic bias whenever the headset is disconnected, to avoid
the unnecessary clicks.

Change-Id: I51795a28d07302c8a2de14479aac93047d20f93f
Signed-off-by: Bhakte, GurudattaX <gurudattax.bhakte@intel.com>
---
 sound/soc/intel/board/cht_bl_dpcm_rt5672.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
index 0e4dfae87fa5..2468be8efbae 100644
--- a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
+++ b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
@@ -115,8 +115,13 @@ static inline void cht_set_codec_power(struct snd_soc_codec *codec,
 	case 0:
 		cht_force_enable_pin(codec, "JD Power", false);
 		cht_force_enable_pin(codec, "Mic Det Power", false);
-		cht_force_enable_pin(codec, "micbias2", false);
-	       break;
+		board_name = dmi_get_system_info(DMI_BOARD_NAME);
+		pr_debug("Turn Off the micbias for %s\n", board_name);
+		if (strcmp(board_name, "Cherry Trail FFD") == 0)
+			cht_force_enable_pin(codec, "micbias1", false);
+		else
+			cht_force_enable_pin(codec, "micbias2", false);
+		break;
 	default:
 		return;
 	}
-- 
1.9.1

